### ISSUE TITLE
"Invalid address" error using a data source for a FHIR server with a SystemAssigned identity

### ISSUE THREAD

**Issue by markrzasa** [hashicorp/terraform-provider-azurerm#21537]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.3.8

### AzureRM Provider Version

3.53.0

### Affected Resource(s)/Data Source(s)

azurerm_healthcare_fhir_service

### Terraform Configuration Files

```hcl
provider "azurerm" {
  features {}
}

locals {
  location      = "eastus2"
  resource_name = join("-", ["eus2", "test", "fhir"])

  fhir_name      = replace(local.resource_name, "-", "")
  workspace_name = replace(local.resource_name, "-", "")
}

data "azurerm_client_config" "current" {}

resource "azurerm_resource_group" "fhir_test" {
  name     = local.resource_name
  location = local.location
}

resource "azurerm_healthcare_workspace" "fhir_test" {
  name                = local.workspace_name
  location            = azurerm_resource_group.fhir_test.location
  resource_group_name = azurerm_resource_group.fhir_test.name
}

resource "azurerm_healthcare_fhir_service" "fhir_test" {
  name                = local.fhir_name
  location            = azurerm_resource_group.fhir_test.location
  resource_group_name = azurerm_resource_group.fhir_test.name
  workspace_id        = azurerm_healthcare_workspace.fhir_test.id
  kind                = "fhir-R4"

  authentication {
    authority = "https://login.microsoftonline.com/${data.azurerm_client_config.current.tenant_id}"
    audience  = "https://${local.workspace_name}-${local.fhir_name}fhir.azurehealthcareapis.com"
  }

  access_policy_object_ids = [
    data.azurerm_client_config.current.object_id
  ]

  identity {
    type = "SystemAssigned"
  }

  cors {
    allowed_origins     = ["*"]
    allowed_headers     = ["*"]
    allowed_methods     = ["GET"]
    max_age_in_seconds  = 3600
    credentials_allowed = false
  }
}

data "azurerm_healthcare_fhir_service" "fhir_test" {
  name         = azurerm_healthcare_fhir_service.fhir_test.name
  workspace_id = azurerm_healthcare_workspace.fhir_test.id
}

output "fhir_data_identity" {
  value = data.azurerm_healthcare_fhir_service.fhir_test.identity
}
```

### Debug Output/Panic Output

```shell
│ Error: setting `identity`: Invalid address to set: []string{"identity", "0", "identity_ids"}
│ 
│   with data.azurerm_healthcare_fhir_service.fhir_test,
│   on main.tf line 60, in data "azurerm_healthcare_fhir_service" "fhir_test":
│   60: data "azurerm_healthcare_fhir_service" "fhir_test" {
```

### Expected Behaviour

The azurerm_healthcare_fhir_service data source is read.

### Actual Behaviour

The following error occurs:

│ Error: setting `identity`: Invalid address to set: []string{"identity", "0", "identity_ids"}
│ 
│   with data.azurerm_healthcare_fhir_service.fhir_test,
│   on main.tf line 60, in data "azurerm_healthcare_fhir_service" "fhir_test":
│   60: data "azurerm_healthcare_fhir_service" "fhir_test" {

### Steps to Reproduce

terraform plan and apply using the configuration file listed above.

### Important Factoids

This worked using version 3.52.0 of the azurerm provider.

**Comment by markrzasa** [comment]
The following pull request has a potential fix for this issue:
https://github.com/hashicorp/terraform-provider-azurerm/pull/21538

**Comment by xiaxyi** [comment]
Thanks @markrzasa for raising this issue and contributing the fix pr. If you don't mind, I'll check your PR and offer some suggestions as the I've worked on this resource before. :)

**Comment by rcskosir** [comment]
Thanks for taking the time to submit this issue. It looks like this has been resolved as of #21594. As such, I am going to mark this issue as closed.
